docs/architecture/lb/nginx.md

```diff
@@ -0,0 +1,270 @@
+
+* TLS PROXY protocol support in SC4S is scheduled for implementation.
+
+Examples for setting up NGINX with the PROXY protocol and DSR are provided below. The Transparent IP method requires complex network configuration. For more details, refer to [this NGINX blog post](https://www.f5.com/company/blog/nginx/ip-transparency-direct-server-return-nginx-plus-transparent-proxy).
+
+
+## Option 1: Configure NGINX with the PROXY Protocol
+
+### Advantages:
+- Easy to set up
+
+### Disadvantages:
+- Available only for TCP, not for UDP or TLS
+- Overwriting the source IP in SC4S is not ideal; the `SOURCEIP` is a hard macro and only `HOST` can be overwritten
+- Overwriting the source IP is available only in SC4S versions greater than 3.4.0
+
+### Configuration
+
+1. On your load balancer (LB) node, add a configuration similar to the following:
+`/etc/nginx/modules-enabled/sc4s.conf`
+```conf
+stream {
+    # Define upstream for each of SC4S hosts and ports
+    # Default SC4S TCP ports are 514, 601
+    # Include your custom ports if applicable
+    upstream stream_syslog_514 {
+        server <SC4S_IP_1>:514;
+        server <SC4S_IP_2>:514;
+    }
+    upstream stream_syslog_601 {
+        server <SC4S_IP_1>:601;
+        server <SC4S_IP_2>:601;
+    }
+
+    # Define a common configuration block for all servers
+    map $server_port $upstream_name {
+        514   stream_syslog_514;
+        601   stream_syslog_601;
+    }
+
+    # Define a virtual server for each upstream connection
+    # Ensure 'proxy_protocol' is set to 'on'
+    server {
+        listen        514;
+        listen        601;
+        proxy_pass    $upstream_name;
+        
+        proxy_timeout 3s;
+        proxy_connect_timeout 3s;
+        
+        proxy_protocol on;
+    }
+}
+```
+
+3. Refer to the NGINX documentation to find the command to reload the service, for example:
+```bash
+sudo nginx -s reload
+```
+
+4. Add the following parameter to the SC4S configuration and restart your instances:
+`/opt/sc4s/env_file`
+```conf
+SC4S_SOURCE_PROXYCONNECT=yes
+```
+
+### Test Your Setup
+Send TCP messages to the load balancer and verify that they are correctly received in Splunk with the host set to your source IP, not the LB's IP:
+
+```bash
+# Test message without IETF frame for port 514/TCP:
+echo "hello world" | netcat <LB_IP> 514
+# Test message with IETF frame for port 601/TCP:
+echo "11 hello world" | netcat <LB_IP> 601
+```
+
+3. Run performance tests based on the [Check TCP Performance](performance-tests.md#check-your-tcp-performance) section.
+
+| Receiver                   | Performance        |
+|----------------------------|--------------------|
+| Single SC4S Server         | 71,738.98 msg/sec  |
+| Load Balancer + 2 Servers  | 99,089.03 msg/sec  |
+
+Please note that load balancer fine-tuning is beyond the scope of the SC4S team's responsibility. For assistance in increasing the TCP throughput of your load balancer instance, contact the NGINX support team.
```

Comment:
Note that load balancer support and fine-tuning is beyond the scope of the SC4S team's responsibility. For assistance in increasing the TCP throughput of your load balancer instance, contact your NGINX support team.